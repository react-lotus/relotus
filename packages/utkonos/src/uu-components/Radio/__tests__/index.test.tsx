/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import type { Ref } from 'react';
import { createRef } from 'react';
import { Radio, RadioProps } from '../index';

const RadioWrapper = (props: Partial<RadioProps> & { customRef?: Ref<HTMLInputElement> }) => {
  const { customRef, ...rest } = props;
  return <Radio value="value" ref={customRef} {...rest} />;
};

describe('Uikit/Radio', () => {
  describe('snapshot', () => {
    it('default', () => {
      const component = <Radio value="value" />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('checked', () => {
      const component = <Radio value="value" checked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled', () => {
      const component = <Radio disabled />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled checked', () => {
      const component = <Radio disabled defaultChecked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('component', () => {
    it('should properly pass ref to the Radio element', () => {
      const ref = createRef<HTMLInputElement>();
      const { getByRole } = render(<RadioWrapper customRef={ref} />);
      ref.current?.focus();
      const inputRadio = getByRole('radio');
      expect(inputRadio).toHaveFocus();
    });
  });
});
