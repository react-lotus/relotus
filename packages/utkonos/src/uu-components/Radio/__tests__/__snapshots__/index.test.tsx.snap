// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uikit/Radio snapshot checked 1`] = `
<label
  className="uu-radio"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    checked={true}
    className="uu-radioInput"
    id="nanoid"
    type="radio"
    value="value"
  />
  <span
    className="uu-radioMark"
  >
    <span
      className="uu-radioMarkPoint uu-radioMarkPoint_checked"
    />
  </span>
</label>
`;

exports[`Uikit/Radio snapshot default 1`] = `
<label
  className="uu-radio"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-radioInput"
    id="nanoid"
    type="radio"
    value="value"
  />
  <span
    className="uu-radioMark"
  >
    <span
      className="uu-radioMarkPoint"
    />
  </span>
</label>
`;

exports[`Uikit/Radio snapshot disabled 1`] = `
<label
  className="uu-radio uu-radio_disabled"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-radioInput"
    disabled={true}
    id="nanoid"
    type="radio"
  />
  <span
    className="uu-radioMark"
  >
    <span
      className="uu-radioMarkPoint uu-radioMarkPoint_disabled"
    />
  </span>
</label>
`;

exports[`Uikit/Radio snapshot disabled checked 1`] = `
<label
  className="uu-radio uu-radio_disabled"
  htmlFor="nanoid"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <input
    className="uu-radioInput"
    defaultChecked={true}
    disabled={true}
    id="nanoid"
    type="radio"
  />
  <span
    className="uu-radioMark"
  >
    <span
      className="uu-radioMarkPoint uu-radioMarkPoint_disabled"
    />
  </span>
</label>
`;
